core-tests/src/test/java/org/mule/runtime/core/api/util/ClassUtilsTestCase.java

```diff
@@ -236,14 +268,78 @@ public void getInheritedFieldValueWithoutRecurse() throws Exception {
+    HashBlob.setStaticHashProperties(hashBlobProperties);
+    List<String> value = ClassUtils.getStaticFieldValue(HashBlob.class, "staticHashProperties", false);
+    assertEquals(hashBlobProperties, value);
+  }
+
+  @Test
+  public void getStaticFieldValueRecursive() throws Exception {
+    final List<String> hashBlobProperties = Arrays.asList(new String[] {"one", "two"});
+    ExtendedHashBlob.setStaticHashProperties(hashBlobProperties);
+    List<String> value = ClassUtils.getStaticFieldValue(ExtendedHashBlob.class, "staticHashProperties", true);
+    assertEquals(hashBlobProperties, value);
+  }
+
+  @Test(expected = NoSuchFieldException.class)
+  public void getStaticFieldValueNotExistentField() throws Exception {
+    ClassUtils.getStaticFieldValue(ExtendedHashBlob.class, "fake", false);
+  }
+
+  @Test(expected = NoSuchFieldException.class)
+  public void getStaticFieldValueNotExistentFieldRecursive() throws Exception {
+    ClassUtils.getStaticFieldValue(ExtendedHashBlob.class, "fake", true);
+  }
+
+
+  @Test(expected = IllegalAccessException.class)
+  public void getStaticFieldValueNotStaticField() throws Exception {
+    ClassUtils.getStaticFieldValue(HashBlob.class, "hash", false);
+  }
+
   @Test
   public void setFieldValue() throws Exception {
-    HashBlob blob = new HashBlob(0);
-    final int hash = hashCode();
+    final int originalHash = hashCode();
+    final int newHash = hashCode();
+    HashBlob blob = new HashBlob(originalHash);
+    ClassUtils.setFieldValue(blob, "hash", newHash, false);
+    assertThat(newHash, equalTo(blob.getHash()));
+  }
 
-    ClassUtils.setFieldValue(blob, "hash", hash, false);
+  @Test
+  public void setFinalFieldValue() throws Exception {
+    final int originalHash = hashCode();
+    final List<String> newFinalHashProperties = Arrays.asList(new String[] {"one", "two"});
+    HashBlob blob = new HashBlob(originalHash);
+    ClassUtils.setFieldValue(blob, "finalHashProperties", newFinalHashProperties, false);
+    assertThat(newFinalHashProperties, equalTo(blob.getFinalHashProperties()));
+  }
 
-    assertThat(hash, equalTo(blob.getHash()));
+  @Test
+  public void setFinalFieldValueRecursive() throws Exception {
+    final int originalHash = hashCode();
+    final List<String> newFinalHashProperties = Arrays.asList(new String[] {"one", "two"});
+    ExtendedHashBlob blob = new ExtendedHashBlob(originalHash);
+    ClassUtils.setFieldValue(blob, "finalHashProperties", newFinalHashProperties, true);
+    assertThat(newFinalHashProperties, equalTo(blob.getFinalHashProperties()));
+  }
+
+  @Test
+  public void setFinalStaticFieldValue() throws Exception {
+    final List<String> newFinalStaticHashProperties = Arrays.asList(new String[] {"one", "two"});
+    ClassUtils.setStaticFieldValue(HashBlob.class, "finalStaticHashProperties", newFinalStaticHashProperties, true);
+    assertThat(newFinalStaticHashProperties, equalTo(HashBlob.getFinalStaticHashProperties()));
```

Comment:
static import

Reply:
Cannot import static method due collision in method name with ExtendedHashBlob class.